[TEST] Talent, Roadmap, RoadmapItem, TalentCategory 테스트 코드 추가

hyeonic opened the pull request:
각 builder 정상적인 예외 처리를 위하여 Assert 클래스 메소드 수정
TalentTest 추가
RoadmapTest 추가
RoadmapItemTest 추가
TalentCategoryTest 추가

#11

hyeonic committed:
test: Talent, Roadmap, RoadmapItem, TalentCategory 테스트 코드 추가

각 builder 정상적인 예외 처리를 위하여 Assert 클래스 메소드 수정
TalentTest 추가
RoadmapTest 추가
RoadmapItemTest 추가
TalentCategoryTest 추가

her0807 commented on src/main/java/com/skhuedin/api/entity/RoadmapItem.java:
```diff
@@ -36,12 +36,17 @@
-        Assert.hasText(String.valueOf(roadmap), "roadmap 필수입니다. ");
-        Assert.hasText(content, "내용 은 필수입니다. ");
-        Assert.hasText(String.valueOf(yearMonth), "yearMonth 필수입니다. ");
+        Assert.notNull(roadmap, "roadmap 필수입니다. ");
```
Assert.notNull 이 있었군요,, 남은 엔티티도 참고해서 작업 진행하겠습니다!

her0807 commented on src/test/java/com/skhuedin/api/entity/TalentTest.java:
```diff
@@ -0,0 +1,239 @@
+        // given
+        User user = createUser();
+
+        String talentImageUrl = "/talent/img";
```
이 부분이 계속 반복되는데,  일반 메소드로 빼서 작성하고, 메소드를 호출하는 형식으로 작성하는 건 어떨까요?


Talent createTalent(User user, String talentImageUrl , String content,  LocalDateTime entranceYear,     
                             LocalDateTime graduationYear, LocalDateTime startCareer){

       return Talent.builder()
                .user(user)
                .talentImageUrl(talentImageUrl)
                .content(content)
                .entranceYear(entranceYear)
                .graduationYear(graduationYear)
                .startCareer(startCareer)
                .build();
}

her0807 reviewed (commented):
스프링 부트 없이 자바 기반으로 테스트 코드를 짜는 좋은 예시를 보여주신 것 같아 많이 참고 하겠습니다!
수고하셨어요 ㅎㅎ 👍  💯

hyeonic committed:
refactor: TalentTest 중복되는 코드를 활용하여 메소드 생성

패키지 이름 변경
api -> skhuedin
entity -> domain

com.skhuedin.api
 -> com.skhuedin.skhuedin

com.skhuedin.api.entity
 -> com.skhuedin.skhuedin.domain

패키지 이름은 후에 스프링 모듈화를 고려하여 좀 더 고민을 보류

her0807 reviewed (approved):
리뷰가 잘 반영된 코드네요. 머지하셔도 좋습니다!
